File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        if not root:
          return 0
        
        left_sum = 0
        stack = [root]
        while len(stack) > 0:
          curr = stack.pop()
          if curr.right:
            stack.append(curr.right)
          if curr.left:
            stack.append(curr.left)
            if not curr.left.left and not curr.left.right:
              left_sum += curr.left.val
              
        return left_sum

File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumOfLeftLeaves = function(root) {
  // iterative approach BFS
  let sum = 0;
  let stack = [root];
  
  while (stack.length > 0) {
    // check for left leaf and add to sum
    const curr = stack.pop();
    if (curr.right) stack.push(curr.right);
    if (curr.left) {
      if (!curr.left.left && !curr.left.right) {
        sum += curr.left.val;
      } else {
        stack.push(curr.left);
      }
    }
  }
  
  return sum;
};